File: nodes/core/market/utils/image_display_node.py
from typing import Any

from core.types_registry import NodeCategory, get_type
from nodes.base.base_node import Base


class ImageDisplay(Base):
    """
    Display node for images. Takes images as input and passes them through for display in the UI.

    - Inputs: 'images' -> Dict[str, str] mapping label to data URL
    - Output: 'images' -> Dict[str, str] mapping label to data URL (pass-through)
    """

    inputs = {
        "images": get_type("ConfigDict") | None,
    }

    outputs = {
        "images": get_type("ConfigDict"),
    }

    CATEGORY = NodeCategory.MARKET

    async def _execute_impl(self, inputs: dict[str, Any]) -> dict[str, Any]:
        images = inputs.get("images")
        if images is None:
            return {"images": {}}

        # Pass through images for display
        return {"images": images}

File: nodes/market/utils/image-display-node.ts
// src/nodes/core/market/utils/image-display-node.ts
// Translated from: nodes/core/market/utils/image_display_node.py

import { Base, NodeCategory, getType, type NodeUIConfig } from '@fig-node/core';

/**
 * Display node for images. Takes images as input and passes them through for display in the UI.
 *
 * - Inputs: 'images' -> Dict[str, str] mapping label to data URL
 * - Output: 'images' -> Dict[str, str] mapping label to data URL (pass-through)
 */
export class ImageDisplay extends Base {
  static inputs = {
    images: getType('ConfigDict'),
  };

  static outputs = {
    images: getType('ConfigDict'),
  };

  static optional_inputs = ['images'];

  static CATEGORY = NodeCategory.MARKET;

  static uiConfig: NodeUIConfig = {
    size: [500, 360],
    resizable: true,
    displayResults: false,
    outputDisplay: {
      type: 'image-viewer',
      bind: 'images',
      options: {
        zoomable: true,
        pannable: true,
        infiniteScroll: true,
        minZoom: 1.0,
        maxZoom: 5.0,
        placeholder: 'No images to display',
      },
    },
  };

  protected async executeImpl(
    inputs: Record<string, unknown>
  ): Promise<Record<string, unknown>> {
    const images = inputs.images;

    if (images === null || images === undefined) {
      return { images: {} };
    }

    // Pass through images for display
    return { images };
  }
}
